public/src/components/common/list_view.jsx

```diff
@@ -51,9 +53,10 @@ const TableListView = (props) => {
                 <TableRow
                   currentUser={currentUser}
                   key={item.id}
+                  auth={auth}
                   is_admin={is_admin}
                   item={item}
-                  handleClick={is_admin ? setCurrentItem : () => {}}
+                  handleClick={setCurrentItem}
```

Comment:
I think it might make sense to have a "handleRedirectClick" and "handleActionClick", since button clicks for all of the list views are either a redirect to a view or edit page, or (in this case with "join") dispatching an action with relevant information. 
This would be useful for the admin user's list too.